pkg/webhooks/admission/jobs/mutate/mutate_job.go

```diff
@@ -219,14 +225,22 @@ func patchDefaultPlugins(job *v1alpha1.Job) *patchOperation {
 		plugins[k] = v
 	}
 
-	// Because the tensorflow-plugin depends on svc-plugin.
+	// Because the tensorflow-plugin and mpi-plugin depends on svc-plugin.
 	// If the svc-plugin is not defined, we should add it.
-	if _, ok := job.Spec.Plugins["tensorflow"]; ok {
+	_, hasTf := job.Spec.Plugins["tensorflow"]
+	_, hasMPI := job.Spec.Plugins[controllerMpi.MPIPluignName]
+	if hasTf || hasMPI {
 		if _, ok := plugins["svc"]; !ok {
 			plugins["svc"] = []string{}
 		}
 	}
 
+	if _, ok := job.Spec.Plugins["mpi"]; ok {
+		if _, ok := plugins["ssh"]; !ok {
+			plugins["ssh"] = []string{}
+		}
+	}
+
```

Comment:
just a suggestion :)I think it better to write the two plugins separately and make comments clear that each plugin depends on different plugins.

Reply:
@Thor-wl Can we discuss how to handle it here?

Reply:
Yes. It depends on the habit of developers, not something matters. You can pick out from our choices. @shinytang6 's suggestion focus on code readability while my suggestion foucus on simplicity of code.

Reply:
Yes, this is more of my personal habit and can also be ignored ）